plugins/destination/postgresql/client/list_tables.go

```diff
@@ -88,22 +88,21 @@ func (c *Client) listTables(ctx context.Context) (schema.Tables, error) {
 			})
 		}
 		table := tables[len(tables)-1]
-		// Note: constraints always have a name in PostgreSQL.
-		// However, we want not only to store the info about the constraint name,
-		// but also the fact that we saw such table.
-		switch pkName {
-		case "":
-			// We still store the fact that we saw the table
-			if _, ok := c.pgTablesToPKConstraints[tableName]; !ok {
-				// Just store the empty string.
-				// This will indicate 2 things:
-				// 1. We saw the table with this name
-				// 2. If this is still empty on insert, the table in the database doesn't have a PK constraint
-				c.pgTablesToPKConstraints[tableName] = ""
+
+		// We always want to record that we saw the table, even if it doesn't have a PK constraint.
+		if _, ok := c.pgTablesToPKConstraints[tableName]; !ok {
+			c.pgTablesToPKConstraints[tableName] = pkConstraintDetails{}
+		}
+		if pkName != "" {
+			if constraint, ok := c.pgTablesToPKConstraints[tableName]; ok {
```

Comment:
You create a copy here & don't update the entry in map, iirc

Reply:
I'm not sure I follow, on line 101 I update the entry in the map

Reply:
Right. What I mean is instead of using the values in the map & doing the overwrite you can use `map-value-pointer` way:
```go
type someStruct struct {
    fields []string
}

aMap := map[string]*someStruct{}

//...

entry, ok := aMap["key"]
if !ok {
    entry := new(someStruct) // or &someStruct{fields:...}
    aMap["key"] = entry
}
// now in ANY case entry is OK to use directly
entry.fields = append(entry.fields, "some", "more", "values")
```

Reply:
done in c21bf852fcdebd57f0fe2341a44ceded16aaad58